### DISCUSSION TITLE
🌴 闲聊区

### DISCUSSION THREAD

**Discussion by VermiIIi0n** [VermiIIi0n/fuckZHS#19] [General]
### 如果有什么想法但是懒得开 issue 的话, 欢迎在此处留言🌴

**Comment by TWLW9784** [comment]
期待更新
![B }I~V{LRSGN(ZZVWGSF{NF](https://user-images.githubusercontent.com/101060038/196437644-16a89a43-263d-4c41-a6da-2892d8912175.png)

**Reply by wDecap** [reply]
![屏幕截图 2023-03-23 233143](https://user-images.githubusercontent.com/74583852/227255048-06a2cbfe-d3ad-4b07-b6d5-09430c25dc29.png)

clone到本地 主方法运行报错 依赖库导了 python3.12版本 求大佬帮帮

**Reply by wDecap** [reply]
已解决pycharm的问题

**Reply by lvhanchao** [reply]
想问一下是怎么解决的

**Reply by VermiIIi0n** [reply]
@lvhanchao 

以前是Windows上用PyCharm直接运行会报错, 但我寻思这问题早修复了啊

**Reply by lvhanchao** [reply]
![屏幕截图 2023-10-30 111732](https://github.com/VermiIIi0n/fuckZHS/assets/66618625/cc235487-11db-4fc1-b539-1d0abe65ea85)
我是不是应该更新一下pychar

**Reply by VermiIIi0n** [reply]
不用, 是我脑瘫, 有次回滚不小心把修复代码滚回去了. 下次更新出来之前先避免用 PyCharm 运行吧...

**Reply by lvhanchao** [reply]
啊？🥴

**Comment by hayeejie** [comment]
![image](https://user-images.githubusercontent.com/96810291/199511123-35c96a90-8e1e-464a-9b9d-f2029b0aaa41.png)
出现这个情况该怎么解决捏

**Reply by VermiIIi0n** [reply]
[这里捏~](https://github.com/VermiIIi0n/fuckZHS/discussions/25)

**Reply by Allencc5658** [reply]
二维码显示乱码怎么解决呢？

**Reply by coke20021215** [reply]
这个问题解决没

**Reply by GodLeaveMe** [reply]
用手机端看一个视频解决验证码就好

**Comment by Alexxuy** [comment]
救命！纯小白新人，啥玩意儿不懂，python已下好，该如何开始部署，已是不知所措！求各路大佬帮助，万分感谢！！！

**Reply by VermiIIi0n** [reply]
啊最近没怎么上号(

文档里写的挺详细的了, 基本上运行 main.py 就行, 如果遇到问题可以去 #25 那儿看看

**Comment by coke20021215** [comment]
最近的登录，发现出现一个问题，刷上一段时间就会闪退

**Reply by coke20021215** [reply]
对于这种问题，如何解决哩

**Reply by VermiIIi0n** [reply]
啊, 这个建议开个 issue 呢, 这么描述我也不知道是什么情况🥹

**Reply by coke20021215** [reply]
脚本运行一段时间，就会莫名的停止运行

**Reply by love768952693** [reply]
我也有这样的问题

**Reply by love768952693** [reply]
> 啊，这个建议开个issue呢，这个描述我也不知道是什么情况🥹

logs文件里面是空的，所以不清楚为啥脚本就退出了。

**Comment by YoumyTheGoust** [comment]
今天更新了啥？

**Reply by VermiIIi0n** [reply]
@jiajiu123 在 PR #36 加入的微信通知推送😁

**Reply by TWLW9784** [reply]
咋使用哇

**Reply by jiajiu123** [reply]
在[pushplus](https://www.pushplus.plus/)里注册一个账号，然后在config里启用并填入token就可以了

**Comment by YoumyTheGoust** [comment]
作者你在tg上有找人推广吗，我在tg上看见有人在卖软件和源码
![image](https://user-images.githubusercontent.com/63649572/202484803-28bd5345-c22c-4980-97e0-9dbd8c9cc653.png)
![image](https://user-images.githubusercontent.com/63649572/202484919-886f7937-2f80-4261-a342-d04d5bebbdab.png)
![image](https://user-images.githubusercontent.com/63649572/202485047-cdccbbb4-edd0-4584-9182-7372dfa2b570.png)

**Reply by VermiIIi0n** [reply]
没有，都开源我怎么可能卖呢哈哈，谢谢告知，我明天更新文档说明下情况。

希望你能在群组里帮忙澄清一下。毕竟被坑的大概也看不到文档。

槽点太多了，但注意他这个打包成exe了，相当危险，希望看到这条消息的人别去尝试运行。

**Reply by YoumyTheGoust** [reply]
okk

**Reply by B1gM8c** [reply]
有趣的是这个打包版本也是免费的，被人拿去卖只能说还是存在信息差😂

**Reply by VermiIIi0n** [reply]
哈哈哈 那看来还是高估这家伙了, 以为至少是他自己打的包🫠

**Comment by CatzTimes** [comment]
![7HQCHAUO%KQ1R)B `4CN3{L](https://user-images.githubusercontent.com/62692767/202891228-f36f4b76-350e-4dbe-ae64-8281ec7d0dfa.png)
这是什么问题
@VermiIIi0n

**Reply by VermiIIi0n** [reply]
中途改变了窗口尺寸吧, 这个进度条的实现挺随便的, 运行时缩小了宽度退格就失效了

**Reply by CatzTimes** [reply]
好吧好吧，无伤大雅，咱这个可以自动答题吗

**Reply by Miku-039** [reply]
单元测试不行, 弹题可以

**Reply by abai2020** [reply]
支持账号密码登录了吗

**Reply by 1837620622** [reply]
支持账号密码登录了吗

**Reply by VermiIIi0n** [reply]
没有

**Comment by cro3odile** [comment]
**(total root chapters: 0)解决方案**
![微信截图_20221215103135](https://user-images.githubusercontent.com/63222351/207758828-eeca0467-ead4-411a-82f9-98f207d0603f.png)
`python main.py -f`
拉取课程清单
然后复制id就好
![微信截图_20221215103217](https://user-images.githubusercontent.com/63222351/207759090-08f09037-2eec-4079-b427-cb27fcdf42fc.png)

**Comment by love768952693** [comment]
用了大佬的软件帮忙学习后，再次使用网页端打开页面。
发现学习的进度在视频详情页全部为0了，只有统计页能看到视频进度。。。。
有人知道怎么处理不。。。。

**Reply by VermiIIi0n** [reply]
抱歉之前没看到，重新登陆网页应该就好了，问题解决了吗？

**Comment by gaotianchi** [comment]
请问这个脚本能否自动识别并跳过已经看完的课程？

**Reply by VermiIIi0n** [reply]
会的，也可以手动指定只刷哪些。

**Comment by gaotianchi** [comment]
<img width="496" alt="image" src="https://user-images.githubusercontent.com/105527253/227697871-eb278621-24a4-4847-81aa-16a5e913023e.png">
在运行程序的情况下又在浏览器上登录智慧树账号，则会出现以上情况，应该是，不能在计算机上同时登录同一个账号。

**Reply by VermiIIi0n** [reply]
是的, 智慧树不允许多个浏览器会话同时登录

**Comment by gaotianchi** [comment]
在使用默认速度 1.5 的情况下，`-l` 参数配置的时间限制长度和实际长度有出入，误差在 5 - 15 分钟左右。

**Reply by VermiIIi0n** [reply]
因为课中途被掐断会让我很不好受, 所以是设计成时限到了也要刷完当节课才结束🫠

**Reply by gaotianchi** [reply]
原来如此

**Comment by NatsumeOnMars** [comment]
请问能否调成默认展示课程数，并且在视频树内展示video_id呢？获取video_id好难  我在课程页面好像找不到  尝试调用fucker的api也经常报错。

**Reply by VermiIIi0n** [reply]
这些在info日志里容易找. 

这屎山代码有时间我重构下, 获取 `video_id` 需要多次调 API, 挺麻烦的. 有 `RecruitAndCourse` ID 的情况下从[这](https://github.com/VermiIIi0n/fuckZHS/blob/cefc049ec00952770c52c7bd1e2371a771c2574e/fucker.py#L302)开始往下照抄应该没问题.

**Reply by NatsumeOnMars** [reply]
好的好的，太感谢了

**Comment by NatsumeOnMars** [comment]
另外提一个建议，请问能否新增一个自动获取cookie并保留cookie在配置文件里的功能，如果cookie有效则跳过扫码登录，这样很方便调试程序，不然每次修改好测试都要扫码（很容易出现二维码已过期的bug）。 把cookie写死在代码里后续修改会很麻烦。

**Reply by VermiIIi0n** [reply]
有道理, 毕竟扫码登录是妥协产物, 没怎么优化体验. 但智慧树 cookies 过期奇快, 正常用倒也区别不大.

*_Edit_*: 在 v2.5.0 更新了

**Reply by VermiIIi0n** [reply]
要自定义 `cookies.json` 只需确保有 `name` 和 `value` 就可以了 (~大概...~)

**Comment by NatsumeOnMars** [comment]
请问自动跳过已看视频的代码在哪哇，找了半天找不到了，我打算做个开关。（因为我发现重复刷也可以拿习惯分，我这个课习惯分很重）。
之前想通过指定视频ID的方式去实现，但发现太复杂了……

**Reply by VermiIIi0n** [reply]
这个需求好怪...

不过观看时间和状态是从智慧树 API 返回的, 把[这两个](https://github.com/VermiIIi0n/fuckZHS/blob/d204e001e79e61e2c4621af82644d8d40db6840e/fucker.py#L419)改成0应该就能重刷了...吧

我不保证能用, 因为进度提交的那套加密算法需要记录观看时间, 这是从未设想过的情况🫠

**Reply by NatsumeOnMars** [reply]
BTW 试了[`video_ids`](https://github.com/VermiIIi0n/fuckZHS/blob/cefc049ec00952770c52c7bd1e2371a771c2574e/fucker.py#L425)、[`video_id`](https://github.com/VermiIIi0n/fuckZHS/blob/cefc049ec00952770c52c7bd1e2371a771c2574e/fucker.py#L599)，作为`-v`的参数都不是真正的id，没办法fuck，会直接跳到`main.py`的最后一块not fucked执行。

**Reply by NatsumeOnMars** [reply]
原来是我把原理想得太简单了

**Reply by NatsumeOnMars** [reply]
刚刚试了一下  确实不可用，会出现 `failed code 8:学习总时长下降了`🤣

**Comment by YsGqHY** [comment]
有没有大佬用这个刷了大多数的共享课了呢，因为我听到有些人说被智慧树抓到会判不及格，所以还是有点担心的

**Reply by Nagi-ovo** [reply]
军理、四史什么的都用它刷的，这个从原理来讲属于比较安全的，用就别怕，怕就别用😹

**Comment by gaotianchi** [comment]
有这样的需求：在程序运行的时候突然有事需要中断网络连接，中断网络后程序保留运行位置，下次运行的时候从中断点处开始。

**Reply by VermiIIi0n** [reply]
现在就是会自动记录最后观看位置的(会由智慧树 API 返回), 只需要在 `config.json` 里把 `save_cookies` 改为 `true` 就可以在`cookies` 未过期的情况下自动恢复了.

**Comment by zfdx123** [comment]
不是很理解他是怎么触发视频记录的，是视频百分比还是什么，我通过Object.defineProperty，干掉了他的加速检测，但是视频播发完成他的百分比依然没有走完

**Reply by zfdx123** [reply]
视频记录指的是触发 saveDatabaseIntervalTimeV2 的相关请求

**Reply by VermiIIi0n** [reply]
以前是每隔30s或者暂停播放就会上传一次进度, 好久没看了不确定变没变.

**Reply by zfdx123** [reply]
我们能不能在控制台手动触发他，让他保存进度呢，他混淆属实离谱，看得难受

**Reply by VermiIIi0n** [reply]
手动触发就是暂停就完了, 虽然没试过但用webdriver点暂停键应该就OK, 不过为什么想要这样做呢, 80%左右以上的进度就会标记成已完成了呀

**Reply by zfdx123** [reply]
我在想用油猴先hook掉playbackrate然后计算出视频进度到达85%作业自动暂停保存进度

**Reply by VermiIIi0n** [reply]
这样啊, 不过真亏你有耐心翻那些混淆的屎山🤯, 我反正是再也不想碰它前端代码了.

**Reply by zfdx123** [reply]
哦他发送的不是视频进度时间而是new 了一个系统时间出来来判断视频到哪了应该是，我这种方法视频虽然加速了但是边上百分比不会变

**Reply by VermiIIi0n** [reply]
其实它还有记录一个叫 watchpoint 的东西, 估计是用来识别异常行为, 这玩意儿每两秒记录一次上次上传进度以来视频进度新增的秒数, 如果暂停视频的话也一直记录, 就是个 `setInterval` 之后运行的玩意儿. 这意味着如果每两项之间的差大于一定限度, 比如三秒, 就是异常的进度了.

我不确定他本地是怎么实现进度计算的, 但总之上传的确实是视频进度, 这在解包之后可以看出来.

**Comment by TheRangGlory** [comment]
python系列都下载了，电脑白痴，这东西怎么用啊
![image](https://github.com/VermiIIi0n/fuckZHS/assets/149854201/ed0a7b22-179b-4f9b-a760-3067315f2c3b)

**Reply by TheRangGlory** [reply]
![image](https://github.com/VermiIIi0n/fuckZHS/assets/149854201/f563d11d-4e79-483c-91ed-12ba91be3224)

**Reply by TheRangGlory** [reply]
用py3.12直接打开会闪退

**Comment by OAT233** [comment]
有考虑整个docker镜像嘛（bushi

**Comment by youngwomen123** [comment]
可以刷规范学习时间吗

**Comment by beef-potato** [comment]
感谢作者，真的帮我省了很多时间 <3

**Comment by sherry204241114** [comment]
> ### 如果有什么想法但是懒得开 issue 的话， 欢迎在此处留言 🌴

电脑小白有没有人教一下怎么用，留个联系方式呗

**Comment by jiazaiwanbi** [comment]
感谢作者，节省了好多好多时间

**Comment by luoying-zero** [comment]
想请问下这个脚本可以刷见面课的回放吗？

**Reply by VermiIIi0n** [reply]
抱歉并不能🫠

**Comment by luoying-zero** [comment]
调用videoList函数，得到
![Screenshot_20240418_104748_edit_11645106963004](https://github.com/VermiIIi0n/fuckZHS/assets/105333299/73a41d9d-0902-43a1-a9de-e7fbf80e3e20)
![Screenshot_20240418_104936](https://github.com/VermiIIi0n/fuckZHS/assets/105333299/b85071c2-eeee-4f74-a500-23ecbe0ef28a)
这是怎么回事？

**Reply by VermiIIi0n** [reply]
智慧树的大便 API 大变过一次，你要找的是图里 `id` 后面的值，但 `-v` 参数没什么人用，我也没怎么测试，不知道还能不能用。

**Comment by HanyesGao** [comment]
![{33E0BD5B-95B8-4951-9367-A32F090E59E0}](https://github.com/VermiIIi0n/fuckZHS/assets/148737798/a51595e9-283b-4427-944b-a2c8159083ea)

**Reply by HanyesGao** [reply]
这种怎么搞

**Reply by HanyesGao** [reply]
已解决 把os换成了shutil

**Comment by Desperadodo** [comment]


**Reply by VermiIIi0n** [reply]
贵校有点严格嘛, 原因的话确实最可能是倍速播放, 有些课程是禁止倍速播放的, 但是 API 里看不出来, 而且最多给个警告所以我没特别处理.

另外这脚本还能跑的情况下我也没注意智慧树有什么更新, 还有可能是API变动造成被识别, 如果惩罚这么严重的话建议先别使用了, 我之后闲下来看看有什么潜在可能.

建议就以用了插件倍速播放视频争取减刑吧, 反正总之还是看了课程, 听起来会好点, 祝好🙏.

**Comment by czx0222** [comment]
扫了二维码没反应

**Comment by amazcuter** [comment]
![image](https://github.com/VermiIIi0n/fuckZHS/assets/90633247/4ae51e32-237b-4c0b-8e27-844eba121cd4)
干到一半出这个了，然后就退出了，重新登录网页要图形验证码。woc我不会被查到吧。
重新开了一遍又继续了。

**Reply by zfdx123** [reply]
我们能不能拿到Captcha的图片？如果是拖动的验证码，我们能不能识别出来拖动多大距离直接返回给后端坐标，如果是字母什么的，让用户输入验证码内容 @VermiIIi0n

**Comment by luoying-zero** [comment]
想问下最近更新新增的`openai`模块有什么用吗？

**Reply by VermiIIi0n** [reply]
可以刷新出的AI课程👍

**Comment by zsyeh** [comment]
感谢作者的贡献代码,节约了大量的生命,原本想自己写一个的,结果发现已经有了,另外想补充一点,那些不会部署的朋友,请耐心去学一点,开源社区,如果你不能证明是源码的问题,请不要问一些没有意义的问题

**Comment by Wu-HZ** [comment]
谢谢！作者的自述文件让我学到了许多新东西（包括间谍过家家（））

**Comment by lazyfreeman** [comment]
在 Termux 上运行脚本，会跳转到手机相册，返回后终端输出扫描二维码的提示但不输出二维码😭

**Reply by lazyfreeman** [reply]
刚发现可以输出成文件，我的错😭

**Reply by Wu-HZ** [reply]
所以直接用termux也能运行么，没有套个linux啥的？

**Reply by lazyfreeman** [reply]
> 所以直接用termux也能运行么，没有套个linux啥的？

直接用termux。不过安装依赖 maturin 时会报错，网上搜一下解决方案就行。

**Reply by Wu-HZ** [reply]
> > 所以直接用termux也能运行么，没有套个linux啥的？
> 
> 直接用termux。不过安装依赖 maturin 时会报错，网上搜一下解决方案就行。

真好，这下电脑不在也能 学习 了

**Reply by luoying-zero** [reply]
呃只是这脚本需要安装openai依赖项。termux里面需要安装rust编译器来构建这个模块，比较花时间🤔

**Reply by luoying-zero** [reply]
不知道openai是不是必需安装的🤔

**Reply by VermiIIi0n** [reply]
不用ai课确实没必要, 有空了我把相关 modules 改成 lazy-load 吧, 不过鉴于 requirements.txt 不支持可选依赖, 想要避免安装相关依赖需要自己删去对应的行

**Comment by Wu-HZ** [comment]
一个课程刷完了，但每日未达标，该如何重来？为了习惯分，每日刷>25分数就够了，保险起见我每次定时40分钟手动关闭（因为每在醒目位置看到如何设置），昨天忘记关了给全刷完了，再运行main.py就不再f*k视频了。我正在查看readme以寻求解答

**Reply by NetheriteBowl** [reply]
看一下Readme.md的line 213，有设置方法

**Comment by niechy** [comment]
#106 看着这个pr挺好的怎么自己关掉了，能不能再拿来用呀
自动过验证码省心多了

**Comment by 6short** [comment]
大佬，怎么设置刷课时长和刷课速度，我启动mian.py就直接开始刷了，终端根本操作不了，😭求大佬解答

**Reply by luoying-zero** [reply]
![Screenshot_20250311_113148](https://github.com/user-attachments/assets/e205cfaf-f7c0-425d-9229-87851228021e)
`-l`选项是单节课刷课时长，`-s`选项是速度，

**Reply by 6short** [reply]
> ![Screenshot_20250311_113148](https://github.com/user-attachments/assets/e205cfaf-f7c0-425d-9229-87851228021e)
> `-l`选项是单节课刷课时长，`-s`选项是速度，

大佬，这个参数444是什么意思😃，求大佬解答

**Reply by luoying-zero** [reply]
444倍速😇

**Reply by luoying-zero** [reply]
呃我也不是大佬。README里面有说明命令行参数怎么用的，你可以看看

**Reply by niechy** [reply]
这速度太夸张了吧，智慧树查不出来吗

**Reply by luoying-zero** [reply]
> 这速度太夸张了吧，智慧树查不出来吗

应急可以试试罢

**Reply by luoying-zero** [reply]
不知道一般的网速能不能支持这个速度🤔

**Comment by fw1688** [comment]
怎么切换账号呀，由于cookies还是啥原因每次运行自动登录自己的号没有二维码，想帮别人刷咋弄呢

**Reply by luoying-zero** [reply]
源代码复制几份放在不同的文件夹，每个文件夹负责一个账号。
或者在main.py:183处
```python
cookies_path = getRealPath("./cookies.json")
```
可以修改`cookies_path`路径，例如设置传参指定`cookies.json`的路径啥的

**Comment by SrjiaChen** [comment]
<img width="2521" height="1339" alt="123" src="https://github.com/user-attachments/assets/8176f710-622c-4414-955f-98e1f5907b2e" />
二维码好大扫不了怎么办

**Reply by xmcaicaizi** [reply]
按住ctrl使用鼠标滚轮放缩？

**Comment by jinkyky** [comment]
造福人类

**Comment by jim-orange** [comment]
<img width="745" height="389" alt="111" src="https://github.com/user-attachments/assets/40558a4f-b5a5-4049-b2f3-65bc7b978c9e" />
<img width="745" height="389" alt="222" src="https://github.com/user-attachments/assets/4d307fe9-f27b-4f97-8f61-8754ed1acc62" />
大佬们 这个401是啥意思呀？

**Comment by keaident-su** [comment]
现在这个项目没有dockerfile，要是能在docker下运行那该多好啊(https://github.com/VermiIIi0n/fuckZHS/issues/131)

**Comment by LGzv666** [comment]
大佬们求教，我想用pushplus推送   |##Captcha required�这个问题，以至于我能及时看到要验证及时上号验证，要不然睡觉的时候老因为这个问题睡不着觉

**Reply by LGzv666** [reply]
因为不会pushplus推送

**Reply by zfdx123** [reply]
找个地方封装一下pushplush的代码，[pushplus文档](https://www.pushplus.plus/doc/guide/demo.html#python%E4%BB%A3%E7%A0%81%E7%A4%BA%E4%BE%8B) 然后打印的时候调用一下就好了
